src/main/java/au/org/ala/names/index/NameSource.java

```diff
@@ -0,0 +1,342 @@
+package au.org.ala.names.index;
+
+import au.org.ala.vocab.ALATerm;
+import org.apache.commons.collections.MapUtils;
+import org.gbif.api.model.registry.Citation;
+import org.gbif.api.model.registry.Contact;
+import org.gbif.api.vocabulary.Country;
+import org.gbif.dwc.terms.*;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * An abstract source of names.
+ * <p>
+ * Subclasses can be used to load name information into a taxonomy.
+ * </p>
+ * <p>
+ * Required and additional fields are derived from
+ * https://tools.gbif.org/dwca-validator/extensions.do
+ * </p>
+ *
+ * @author Doug Palmer &lt;Doug.Palmer@csiro.au&gt;
+ * @copyright Copyright &copy; 2017 Atlas of Living Australia
+ */
+abstract public class NameSource {
+    /** Fields expected in the DwCA */
+    protected static final List<Term> TAXON_REQUIRED = Arrays.asList(
```

Comment:
This needs reworking as a configurable thing, as it got more complicated than I expected. For now, though, I'd like to push doing that to a later version, since it's alright on the night. There is a fairly static set of elements that must be present and a user-friendly way of ordering things.